lib/SPIRV/SPIRVUtil.cpp

```diff
@@ -836,14 +827,14 @@ parseNode(Module *M, const llvm::itanium_demangle::Node *ParamType,
   return PointeeTy ? TypedPointerType::get(PointeeTy, AS) : nullptr;
 }
 
-void getParameterTypes(Function *F, SmallVectorImpl<TypedPointerType *> &ArgTys,
+bool getParameterTypes(Function *F, SmallVectorImpl<Type *> &ArgTys,
                        std::function<std::string(StringRef)> NameMapFn) {
   using namespace llvm::itanium_demangle;
   // If there's no mangled name, we can't do anything. Also, if there's no
   // parameters, do nothing.
   StringRef Name = F->getName();
   if (!Name.startswith("_Z") || F->arg_empty())
-    return;
+    return Name.startswith("_Z");
```

Comment:
Not sure if I understand, this line. We don't want to just return `false `in case if `F->arg_empty()`, but why?

Reply:
Some builtins take no arguments, so if the function was a `_Z3foov`, we don't want to say it was unsuccessfully demangled.

Reply:
Thanks for the explanation!